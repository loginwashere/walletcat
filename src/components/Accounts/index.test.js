import React from 'react'
import renderer from 'react-test-renderer'
import {
  accountSeeder,
  userCurrencySeeder,
  currencySeeder
} from '../../../server/seeds'
import { Accounts } from '.'

describe('components:Accounts:', () => {
  const dispatch = jest.fn()
  const page = 1
  const account = accountSeeder.items[0]
  const accounts = { [account.id]: account }
  const pages = { [page]: { ids: [account.id] } }
  const userCurrencies = userCurrencySeeder.items
    .reduce((result, current) => ({ ...result, [current.id]: current }), {})
  const currencies = currencySeeder.items
    .reduce((result, current) => ({ ...result, [current.id]: current }), {})

  it('renders without crashing', () => {
    const tree = renderer.create(
      <Accounts accounts={accounts}
                pages={pages}
                page={page}
                currencies={currencies}
                userCurrencies={userCurrencies}
                dispatch={dispatch} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
