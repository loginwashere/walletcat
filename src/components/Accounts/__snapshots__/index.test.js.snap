exports[`components:Accounts: renders without crashing 1`] = `
<div>
  <h1>
    Accounts
    <button
      action="push"
      className="pull-right btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      Create
    </button>
  </h1>
  <div
    className="list-group">
    <a
      className="list-group-item"
      onClick={[Function]}
      style={Object {}}>
      <span
        className="badge badge-danger pull-right">
        0.00000000
         
        AED
      </span>
      <h4
        className="list-group-item-heading truncate">
        Wallet
      </h4>
      <p
        className="list-group-item-text truncate">
        My Wallet
      </p>
    </a>
  </div>
  <ul
    className="pager">
    <li
      className="disabled previous"
      style={undefined}>
      <a
        action="push"
        href="#"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "pointerEvents": "none",
          }
        }
        tabIndex={-1}>
        Previous Page
      </a>
    </li>
    <li
      className="disabled next"
      style={undefined}>
      <a
        action="push"
        href="#"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "pointerEvents": "none",
          }
        }
        tabIndex={-1}>
        Next Page
      </a>
    </li>
  </ul>
</div>
`;
